### PR TITLE
srun jobID docs

### PR DESCRIPTION
**Pull request by nikishe**

### CODE REVIEW

File: hpc/scheduler/advanced_job_analysis_and _profiling.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+
+Accessing A Running Multi Node Slurm Batch job
+==============================================
+
+In the scenario you are running a multi node slurm job you can use ``squeue`` to see the nodes your job is using:
+
+.. code-block:: shell
+
+    squeue --user=$USER
```
Comment (willfurnass) [inline hpc/scheduler/advanced_job_analysis_and _profiling.rst:63]:
```suggestion
    squeue --me
```

File: hpc/scheduler/advanced_job_analysis_and _profiling.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    ps
+
+    nvidia-smi
+
+    top
+
+    lsof
+
+
+Accessing A Running Multi Node Slurm Batch job
```
Comment (willfurnass) [inline hpc/scheduler/advanced_job_analysis_and _profiling.rst:56]:
```suggestion
Accessing A Running Multi-Node Slurm Batch job
```

File: hpc/scheduler/advanced_job_analysis_and _profiling.rst [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+==============================================
+
+In the scenario you are running a multi node slurm job you can use ``squeue`` to see the nodes your job is using:
```
Comment (willfurnass) [inline hpc/scheduler/advanced_job_analysis_and _profiling.rst:59]:
```suggestion
In the scenario you are running a multi-node Slurm job you can use ``squeue`` to see the nodes your job is using:
```

File: hpc/scheduler/advanced_job_analysis_and _profiling.rst [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+.. code-block:: shell
+
+    squeue --user=$USER
+
+**Example output:**
+
+.. code-block:: shell
+
+    squeue --user=$USER
```
Comment (willfurnass) [inline hpc/scheduler/advanced_job_analysis_and _profiling.rst:69]:
```suggestion
    squeue --me
```

File: hpc/scheduler/advanced_job_analysis_and _profiling.rst (hunk 1 above)
Comment (jkwmoore) [reply hpc/scheduler/advanced_job_analysis_and _profiling.rst:63]: Agree with this, I believe we also have quite a lot of the same change to make in the job submission pages.

Will look at this now.

File: hpc/scheduler/advanced_job_analysis_and _profiling.rst [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+    For majority of the users the :ref:`job_submission_control` page and the output log files you produce should suffice in analysing the perfomance of a batch job. This section is for users who want an even more granular view of the perfomance of their batch jobs by observing what is happening on the node their job is running. 
+
+.. warning::
+
+    Since your job has a fixed set of  memory and CPU resources, carrying out resource hungry operations might lead to the scheduler killing your batch job due to errors like ``out of memory``. 
+
+    Abuse of this feature to carry out tasks that are not profiling and perfomance analysis of your running batch job might lead to your account being suspended.
+
+Accessing A Running Single Node Slurm Batch job
```
Comment (willfurnass) [inline hpc/scheduler/advanced_job_analysis_and _profiling.rst:16]:
```suggestion
Accessing A Running Single-Node Slurm Batch job
```

File: hpc/scheduler/advanced_job_analysis_and _profiling.rst [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+===============================================
+
+In some cases, you might want to access the resources of an existing scheduler allocation, for example a single node batch job that is in the running state. After allocating resources with ``sbatch`` you can access the resources (node) of this allocation with:
```
Comment (willfurnass) [inline hpc/scheduler/advanced_job_analysis_and _profiling.rst:19]: How about this instead?

```suggestion
In some cases, you might want to interact with a batch job in the ``RUNNING``  state (e.g. for fault-finding, debugging or profiling purposes).  You can start an interactive session within the resource allocation (memory and CPU cores on particular nodes) associated with the job with:
```

File: hpc/scheduler/advanced_job_analysis_and _profiling.rst [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+.. code-block:: shell
+
+    srun --jobid=<JOBID> --pty /usr/bin/bash
+
+The command creates a new job step and starts interactive bash shell session within a previously allocated job, allowing you to interact with the resources allocated to that job.
```
Comment (willfurnass) [inline hpc/scheduler/advanced_job_analysis_and _profiling.rst:25]:
```suggestion
The command creates a new Job Step in the batch job with ID ``<JOBID>`` and starts interactive bash shell session within a that Job Step, allowing you to interact with the resources allocated to that job.
```

File: hpc/scheduler/advanced_job_analysis_and _profiling.rst [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+If all the allocated CPU resources are already used, srun will prohibit the new job step the access to the resources. However, the argument ``--overlap`` can be passed to srun to allow job steps to overlap on the resources.
```
Comment (willfurnass) [inline hpc/scheduler/advanced_job_analysis_and _profiling.rst:27]:
```suggestion
If all the allocated CPU resources are already used, ``srun`` will prohibit the new Job Step the access to the resources. However, the argument ``--overlap`` can be passed to ``srun`` to allow Job Steps to overlap on the resources.
```

File: hpc/scheduler/advanced_job_analysis_and _profiling.rst [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+.. code-block:: shell
+
+    scontrol listpids |grep <JOBID>
+
+    or
+
+    ps -u $USER
+
+start profiling and analysing the perfomance of the node and the job by using commands such as:
```
Comment (willfurnass) [inline hpc/scheduler/advanced_job_analysis_and _profiling.rst:43]:
```suggestion
Start profiling and analysing the perfomance of the node and the job by using commands such as:
```

File: hpc/scheduler/advanced_job_analysis_and _profiling.rst (hunk 7 above)
Comment (willfurnass) [reply hpc/scheduler/advanced_job_analysis_and _profiling.rst:25]: Also, do we want to link to reference info that explains what a Job Step is?

**Review by willfurnass — CHANGES_REQUESTED** [review]
Mostly looks good!  A few wording suggestions (and some grammatical pedantry).

File: hpc/scheduler/advanced_job_analysis_and _profiling.rst [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+    For majority of the users the :ref:`job_submission_control` page and the output log files you produce should suffice in analysing the perfomance of a batch job. This section is for users who want an even more granular view of the perfomance of their batch jobs by observing what is happening on the node their job is running. 
+
+.. warning::
+
+    Since your job has a fixed set of  memory and CPU resources, carrying out resource hungry operations might lead to the scheduler killing your batch job due to errors like ``out of memory``. 
+
+    Abuse of this feature to carry out tasks that are not profiling and perfomance analysis of your running batch job might lead to your account being suspended.
+
+Accessing A Running Single-Node Slurm Batch job
```
Comment (Carldkennedy) [inline hpc/scheduler/advanced_job_analysis_and _profiling.rst:16]:
```suggestion
Accessing a Running Single-Node Slurm Batch job
```

File: hpc/scheduler/advanced_job_analysis_and _profiling.rst [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+===============================================
+
+In some cases, you might want to interact with a batch job in the ``RUNNING``  state (e.g. for fault-finding, debugging or profiling purposes).  You can start an interactive session within the resource allocation (memory and CPU cores on particular nodes) associated with the job with:
+
+.. code-block:: shell
+
+    srun --jobid=<JOBID> --pty /usr/bin/bash
+
+The command creates a new Job Step in the batch job with ID ``<JOBID>`` and starts interactive bash shell session within a that Job Step, allowing you to interact with the resources allocated to that job.
```
Comment (Carldkennedy) [inline hpc/scheduler/advanced_job_analysis_and _profiling.rst:25]:
```suggestion
The command creates a new Job Step in the batch job with ID ``<JOBID>`` and starts an interactive bash shell session within that Job Step, allowing you to interact with the resources allocated to that job.
```

File: hpc/scheduler/advanced_job_analysis_and _profiling.rst [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+If all the allocated CPU resources are already used, ``srun`` will prohibit the new Job Step the access to the resources. However, the argument ``--overlap`` can be passed to ``srun`` to allow Job Steps to overlap on the resources.
```
Comment (Carldkennedy) [inline hpc/scheduler/advanced_job_analysis_and _profiling.rst:27]:
```suggestion
If all the allocated CPU resources are already used, ``srun`` will prohibit the new Job Step to access those resources. However, the argument ``--overlap`` can be passed to ``srun`` to allow Job Steps to share those resources.
```

File: hpc/scheduler/advanced_job_analysis_and _profiling.rst [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+    ps
+
+    nvidia-smi
+
+    top
+
+    lsof
+
+
+Accessing A Running Multi-Node Slurm Batch job
```
Comment (Carldkennedy) [inline hpc/scheduler/advanced_job_analysis_and _profiling.rst:56]:
```suggestion
Accessing a Running Multi-Node Slurm Batch job
```

**Review by Carldkennedy — COMMENTED** [review]
A few minor suggestions. Otherwise LGTM

**Review by Carldkennedy — APPROVED** [review]
LGTM

**Pull request merged**